modules/lang-painless/src/test/java/org/elasticsearch/painless/FunctionRefTests.java

```diff
@@ -273,4 +273,22 @@ public void testReturnVoidDef() {
         });
         assertThat(expected.getMessage(), containsString("lambda expects return type [long], but found return type [void]"));
     }
+
+    public void testPrimitiveMethodReferences() {
+        assertEquals(1L, exec("long test(Function s) {return s.apply(Integer.valueOf(1));} return test(int::intValue);"));
+        assertEquals(1L, exec("long test(Supplier s) {return s.get();} int i = 1; return test(i::intValue);"));
```

Comment:
It would be good to test for other primitive types.

Reply:
Done.